### PR TITLE
implement gene

### PR DESCRIPTION
**Pull request by AliNazariii**

### CODE REVIEW

File: final/chromosome.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Chromosome: 
+    def __init__(self, genes):
+        self.genes = genes
+        self.fitness = 0
+
+    def swap(self, first, second):
+        self.genes[first], self.genes[second] = self.genes[second], self.genes[first]
+    
+    def calc_fitness(self):
```
Comment (mrakhgari) [inline final/chromosome.py:9]: we can use @property and change the name of method to `fitness`(see @property in python). for speedup of code we can use `if self.fitness is not 0: return self.fitness`.

File: final/ea.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from abc import ABCMeta, abstractmethod
+
+class EvolutionaryAlgorithmsInterface:
```
Comment (mrakhgari) [inline final/ea.py:3]: I think it's better to have comments.

File: final/main.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import utils
+from tsp import TSP
+
+population_size = 50
+n_generation = 100
+tournament_size = 20
+mutation_rate = 0.01 
+input_path = './tsp_data.txt'   
+
+def main():
```
Comment (mrakhgari) [inline final/main.py:11]: We can get inputs from "sys.argv". Of course, this is a suggestion.

Comment (AliNazariii) [reply final/main.py:11]: yes, you are true.
Done

File: final/ea.py (hunk 2 above)
Comment (AliNazariii) [reply final/ea.py:3]: hmmm yes.
I added a few of them.

File: final/chromosome.py (hunk 1 above)
Comment (AliNazariii) [reply final/chromosome.py:9]: Did you notice that fitness is one of our class properties?
We need a method to calculate fitness.
Could you explain more?

Comment (AliNazariii) [reply final/chromosome.py:9]: I think @property is just a decorator instead of using getter and setter.

**Pull request merged**